frontend/taipy-gui/src/components/Taipy/DateRange.tsx

```diff
@@ -86,7 +94,7 @@ const DateRange = (props: DateRangeProps) => {
         (v: Date | null, start: boolean) => {
             setValue((dates) => {
                 if (v !== null && isValid(v)) {
-                    const newDate = getTimeZonedDate(v, tz, withTime);
+                    const newDate = roundToNearest15Minutes(getTimeZonedDate(v, tz, withTime)); // Round to nearest 15 mins
```

Comment:
this behavior should be optional and commanded by a property.
I propose that this property would be set the value in time format (hh:mm:ss) so that we can define before showing the time selector, which times the user can select
I propose the property name: time_step
In this example, `time_step=00:15` 
or `time_step=01:00` for letting the user select only time on the hour.
This should also apply to date_selector and time_selector